apps/admin-gui/src/app/shared/components/dialogs/copy-sponsored-members-dialog/copy-sponsored-members-dialog.component.ts

```diff
@@ -0,0 +1,124 @@
+import { Component, Inject, OnInit } from '@angular/core';
+import {
+  MembersManagerService,
+  MemberWithSponsors,
+  RichUser,
+  User,
+} from '@perun-web-apps/perun/openapi';
+import { MAT_DIALOG_DATA, MatDialogRef } from '@angular/material/dialog';
+import {
+  NotificatorService,
+  PerunTranslateService,
+  StoreService,
+} from '@perun-web-apps/perun/services';
+import { SelectionModel } from '@angular/cdk/collections';
+import { TABLE_ADD_SPONSORED_MEMBERS } from '@perun-web-apps/config/table-config';
+import { formatDate } from '@angular/common';
+
+export interface CopySponsoredMembersDialogData {
+  voId: number;
+  theme: string;
+  voSponsors: RichUser[];
+  findSponsorsAuth: boolean;
+}
+@Component({
+  selector: 'app-copy-sponsored-members-dialog',
+  templateUrl: './copy-sponsored-members-dialog.component.html',
+  styleUrls: ['./copy-sponsored-members-dialog.component.scss'],
+})
+export class CopySponsoredMembersDialogComponent implements OnInit {
+  loading = false;
+  theme: string;
+  sponsorType: string;
+  tableId = TABLE_ADD_SPONSORED_MEMBERS;
+  voSponsors: RichUser[];
+  sponsorFrom: User;
+  sponsorTo: User;
+  sponsoredMembers: MemberWithSponsors[];
+  filteredSponsoredMembers: MemberWithSponsors[];
+  selection: SelectionModel<MemberWithSponsors> = new SelectionModel<MemberWithSponsors>(true, []);
+  firstSearchDone = false;
```

Comment:
`firstSearchDone` does not really represent what the variable actually checks for and that some "source" sponsor was selected.